components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -256,14 +256,35 @@ abstract class AbstractFetchDownloadService : Service() {
             store.state.downloads[it]
         } ?: return START_REDELIVER_INTENT
 
+        if (intent.action ?: "" == ACTION_REMOVE_PRIVATE_DOWNLOAD) {
+            handleRemovePrivateDownloadIntent(download)
```

Comment:
We should also stop the download entirely here.

Reply:
🤔 We decided to handle that separately as we likely need to prompt the user for this.

Reply:
I filed https://github.com/mozilla-mobile/fenix/issues/18160 for this case